### ISSUE TITLE
Support links to arbitrary items

### ISSUE THREAD

**Issue by Toxaris** [ps-mr/website#31]
@Blaisorblade asks for more `{% member ...%}`-like tags in #30.

**Comment by Blaisorblade** [comment]
Yep, would be cool, though not urgent. The ones for papers seem most important since they target generated links, and even there...

**Comment by Toxaris** [comment]
Instead of adding more special-purpose tags like `{% member ... %}, I'm considering to replace the `{% member ... %}` tag by a more general-purpose tag, maybe `{% ref text %}`, that can be used to insert references to arbitrary targets. It would work as follows:

Firstly, the target of the reference is determined:

 * If the first word of `text` is the id of a page or post on the website, the target is that page or post. (An id is something like `/research/functors`).
 * If the first word of `text` is the key of a bibtex item, the target is that bibtex item.
 * If the first word of `text` is a URL, the target is that website.
 * If the first word is something else, the target is automagically determined based on the whole `text`. 

Secondly, the text of the reference is determined:

 * If the target of the reference was given by the first word of `text`, and there are other words in `text`, the text of the reference are the other words.
 * If the target of the reference was given by the first word of `text`, but there are no other words in `text`, the text of the reference is automagically determined by the target.
 * If the target of the reference was automagically determined based on the whole `text`, the text of the reference is the whole `text`.

Thirdly, fancy extras of the reference are determined:

 * If the target of the reference is a group member, the fancy extras of the current `{% member ... %}` tag are used.
 * If the target of the reference is a bibtex item, the fancy extras are a preview of the bibliographic reference.
 * If the target is something else, there are no fancy extras.

This should allow us to replace all current occurrences of `{% member ... %}` by the new tag, and also start using the new tag for most or all other links.

Note that bibtex items could be drawn from the current `publications.bib` to link to our own publications, but also from a to-be-added `references.bib` to link to other people's publications. This would allow to use the new tag for lists of papers in seminar websites.

**Comment by Toxaris** [comment]
An existing issue with the `{% ... member %}` tag is that the website build breaks if one uses a non-existing member name. Instead of breaking the build, one could imagine generating a broken link.
